browser/ui/views/split_view/split_view_location_bar.cc

```diff
@@ -0,0 +1,236 @@
+}
+
+SplitViewLocationBar::~SplitViewLocationBar() = default;
+
+// static
+SplitViewLocationBar* SplitViewLocationBar::Create(views::View* web_view) {
+  CHECK(web_view->GetWidget());
+  auto* location_bar = new SplitViewLocationBar();
+
+  views::Widget::InitParams params;
+  params.type = views::Widget::InitParams::TYPE_CONTROL;
+  params.parent = web_view->GetWidget()->GetNativeView();
+  params.activatable = views::Widget::InitParams::Activatable::kNo;
+  params.delegate = location_bar;
+  auto* widget = new views::Widget();
+  widget->Init(std::move(params));
+
+  location_bar->view_observation_.Observe(web_view);
+  location_bar->UpdateVisibility();
+  location_bar->UpdateBounds();
+
+  widget->Show();
+
+  return location_bar;
+}
+
+void SplitViewLocationBar::SetWebContents(content::WebContents* web_contents) {
+  if (this->web_contents() == web_contents) {
+    return;
+  }
+
+  Observe(web_contents);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnPaintBackground(gfx::Canvas* canvas) {
+  auto* cp = GetColorProvider();
+  CHECK(cp);
+
+  auto path = GetBorderPath(/*close*/ true);
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorToolbar));
+  flags.setAntiAlias(true);
+  canvas->DrawPath(path, flags);
+}
+
+void SplitViewLocationBar::OnPaintBorder(gfx::Canvas* canvas) {
+  auto* cp = GetColorProvider();
+  CHECK(cp);
+
+  auto path = GetBorderPath(/*close*/ false);
+  cc::PaintFlags flags;
+  flags.setColor(cp->GetColor(kColorBraveSplitViewInactiveWebViewBorder));
+  flags.setAntiAlias(true);
+  flags.setStyle(cc::PaintFlags::kStroke_Style);
+  flags.setStrokeWidth(2);
+  canvas->DrawPath(path, flags);
+}
+
+void SplitViewLocationBar::PrimaryPageChanged(content::Page& page) {
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::WebContentsDestroyed() {
+  Observe(nullptr);
+  UpdateURLAndIcon();
+}
+
+void SplitViewLocationBar::OnViewVisibilityChanged(views::View* observed_view,
+                                                   views::View* starting_view) {
+  UpdateVisibility();
+}
+
+void SplitViewLocationBar::OnViewBoundsChanged(views::View* observed_view) {
+  UpdateBounds();
+}
+
+void SplitViewLocationBar::OnViewIsDeleting(views::View* observed_view) {
+  view_observation_.Reset();
+}
+
+void SplitViewLocationBar::UpdateVisibility() {
+  auto* view = view_observation_.GetSource();
+  if (view && view->GetVisible()) {
+    GetWidget()->ShowInactive();
+  } else {
+    GetWidget()->Hide();
+  }
+}
+
+void SplitViewLocationBar::UpdateBounds() {
+  auto* view = view_observation_.GetSource();
+  if (!view) {
+    return;
+  }
+
+  gfx::Point point;
+  views::View::ConvertPointToWidget(view, &point);
+
+  auto* widget = GetWidget();
+  CHECK(widget);
+  GetWidget()->SetBounds(gfx::Rect(point, GetPreferredSize()));
+}
+
+void SplitViewLocationBar::UpdateURLAndIcon() {
+  if (auto* contents = web_contents()) {
+    GURL url = contents->GetLastCommittedURL();
+    url_->SetText(base::UTF8ToUTF16(url.host()));
```

Comment:
It might be worth using on of the Chromium methods for formatting this:
- `AppBrowserController::FormatUrlOrigin` does something like what we (probably) want here
- Otherwise there's `url_formatter::FormatUrl`

We should probably have the security team look at this - formatting URLs can get pretty complicated - I think Chromium has some logic somewhere for handling lookalike characters (i.e. from cyrillic) and converting unicode to punycode

Reply:
Also great suggestion. Done.